### DISCUSSION TITLE
Make the documentation great again

### DISCUSSION THREAD

**Discussion by lucasbordeau** [remix-run/react-router#10466] [Proposals]
### Describe what's incorrect/missing in the documentation

Here is concise feedback on the current situation which is intolerable for this project size : 

- Feels convoluted and way too complex for the average use case of a BrowserRouter with a few routes.
- No installation page in the getting started section, this is basics.
- Have to go through a full tutorial to get started.
- Examples are on a repository for the user to look, the simplest ones could be copy/pasted on the example page or have a stackblitz/codesandbox demo iframe.
- Overall it feels like the documentation mixes small tips with general concepts in an unstructured way.
- Why can't we have the v5 documentation back ? It was a great and original UI/UX experience !

Proposal : 

- Put an installation copiable component on the home page, above the 4 box section.
- Add an installation page in the getting started section with a copiable section containing the line to install everything needed and other considerations.
- Add a SIMPLE example page just below installation page, to cover 80% of most uses cases which should be how to install a very basic BrowserRouter in your React app the new way vs. the old way (or redirect to migrating from v5 for the old way).
- Picking a router page : just put a table with two columns : Your situation | Recommended routers
- Picking a router page : Put the DataAPI section and updates into a single coherent sub heading
- Example section : add examples from the repository and explain for dummies. Keep advanced examples in another section or in the repository.

**Comment by bryanprimus** [comment]
I really need search functionality, I couldn't find typescript usage section. :)

**Comment by ChuckJonas** [comment]
yea, 100% agree... after searching for the "quick start" and only being able to find the way "over-developed" tutorial, I had to know if others felt the same way.

- Has me install multiple packages without any explanation and still don't know if I actually need `localforage match-sorter sort-by` in my project (are these peer-dependencies)?
- has and discusses `css`. Why?
- has me hook up a custom "not found" page before I even have connected any routes
- overall just way too much text to grok
- (that's as far as I made it).

I guess as a true tutorial it's fine... but a quick start is desperately needed. 

I just want to setup some simple routes with url params and load my data with regular `react-query` hooks (not the "loader").

**Comment by Dron007** [comment]
I agree, absence of examples on the site is very frustrating. It takes too much time to scan all directories in GitHub looking for what you need. These pages are almost useless without examples:
https://reactrouter.com/en/main/router-components/router
https://reactrouter.com/en/main/utils/match-path

V5 documentation was much better.
